---
 kpartx/kpartx.8 |    3 +++
 kpartx/kpartx.c |   13 ++++++++++---
 2 files changed, 13 insertions(+), 3 deletions(-)

Index: multipath-tools/kpartx/kpartx.8
===================================================================
--- multipath-tools.orig/kpartx/kpartx.8
+++ multipath-tools/kpartx/kpartx.8
@@ -26,6 +26,9 @@ List partition mappings that would be ad
 .B \-p
 set device name-partition number delimiter
 .TP
+.B \-f
+force creation of mappings; overrides 'no_partitions' feature
+.TP
 .B \-g
 force GUID partition table (GPT)
 .TP
Index: multipath-tools/kpartx/kpartx.c
===================================================================
--- multipath-tools.orig/kpartx/kpartx.c
+++ multipath-tools/kpartx/kpartx.c
@@ -82,19 +82,22 @@ initpts(void)
 	addpts("sun", read_sun_pt);
 }
 
-static char short_opts[] = "ladgvp:t:su";
+static char short_opts[] = "ladfgvp:t:su";
 
 /* Used in gpt.c */
 int force_gpt=0;
 
+int force_devmap=0;
+
 static int
 usage(void) {
-	printf("usage : kpartx [-a|-d|-l] [-v] wholedisk\n");
+	printf("usage : kpartx [-a|-d|-l] [-f] [-v] wholedisk\n");
 	printf("\t-a add partition devmappings\n");
 	printf("\t-d del partition devmappings\n");
 	printf("\t-l list partitions devmappings that would be added by -a\n");
 	printf("\t-p set device name-partition number delimiter\n");
 	printf("\t-g force GUID partition table (GPT)\n");
+	printf("\t-f force devmap create\n");
 	printf("\t-v verbose\n");
 	printf("\t-s sync mode. Don't return until the partitions are created\n");
 	return 1;
@@ -235,6 +238,9 @@ main(int argc, char **argv)
 	}
 
 	while ((arg = getopt(argc, argv, short_opts)) != EOF) switch(arg) {
+		case 'f':
+			force_devmap=1;
+			break;
 		case 'g':
 			force_gpt=1;
 			break;
@@ -331,7 +337,8 @@ main(int argc, char **argv)
 
 	if (!mapname)
 		mapname = device + off;
-	else if (dm_no_partitions((unsigned int)MAJOR(buf.st_rdev),
+	else if (!force_devmap &&
+		 dm_no_partitions((unsigned int)MAJOR(buf.st_rdev),
 				  (unsigned int)MINOR(buf.st_rdev))) {
 		/* Feature 'no_partitions' is set, return */
 		return 0;
