---
 libmultipath/hwtable.c  |    5 +++--
 multipath.conf.defaults |    5 +++--
 2 files changed, 6 insertions(+), 4 deletions(-)

Index: multipath-tools/libmultipath/hwtable.c
===================================================================
--- multipath-tools.orig/libmultipath/hwtable.c
+++ multipath-tools/libmultipath/hwtable.c
@@ -992,16 +992,17 @@ static struct hwentry default_hw[] = {
 		.vendor        = "NETAPP",
 		.product       = "LUN.*",
 		.getuid        = DEFAULT_GETUID,
-		.features      = "1 queue_if_no_path",
+		.features      = "3 queue_if_no_path pg_init_retries 50",
 		.hwhandler     = DEFAULT_HWHANDLER,
 		.selector      = DEFAULT_SELECTOR,
 		.pgpolicy      = GROUP_BY_PRIO,
 		.pgfailback    = -FAILBACK_IMMEDIATE,
+		.flush_on_last_del = FLUSH_ENABLED,
 		.rr_weight     = RR_WEIGHT_NONE,
 		.no_path_retry = NO_PATH_RETRY_UNDEF,
 		.minio         = 128,
 		.minio_rq      = DEFAULT_MINIO_RQ,
-		.checker_name  = DIRECTIO,
+		.checker_name  = TUR,
 		.prio_name     = PRIO_ONTAP,
 	},
 	/*
Index: multipath-tools/multipath.conf.defaults
===================================================================
--- multipath-tools.orig/multipath.conf.defaults
+++ multipath-tools/multipath.conf.defaults
@@ -481,14 +481,15 @@
 #		vendor			"NETAPP"
 #		product			"LUN.*"
 #		getuid_callout		"/lib/udev/scsi_id --whitelisted --device=/dev/%n"
-#		features		"1 queue_if_no_path"
+#		features		"3 queue_if_no_path pg_init_retries 50"
 #		hardware_handler	"0"
 #		path_selector		"round-robin 0"
 #		path_grouping_policy	group_by_prio
 #		failback		immediate
 #		rr_weight		uniform
 #		rr_min_io		128
-#		path_checker		directio
+#		path_checker		tur
+#		flush_on_last_del	yes
 #		prio			ontap
 #	}
 #	device {
