---
 libmultipath/checkers/tur.c |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

Index: multipath-tools/libmultipath/checkers/tur.c
===================================================================
--- multipath-tools.orig/libmultipath/checkers/tur.c
+++ multipath-tools/libmultipath/checkers/tur.c
@@ -428,10 +428,9 @@ libcheck_check (struct checker * c)
 		if (ct->thread) {
 			/* pthread cancel failed. continue in sync mode */
 			pthread_mutex_unlock(&ct->lock);
-			condlog(3, "%d:%d: tur thread not responding, "
-				"using sync mode", TUR_DEVT(ct));
-			return tur_check(c->fd, c->timeout, c->message,
-					 ct->wwid);
+                        condlog(3, "%d:%d: tur thread not responding ",
+                                TUR_DEVT(ct));
+                        return PATH_DOWN;
 		}
 		/* Start new TUR checker */
 		ct->state = PATH_UNCHECKED;
